### PR TITLE
Fixes #27 - Fix failing test_that_user_can_edit_profile

### LINKED ISSUE #27: Fix test_that_user_can_edit_profile
08:16:41 [gw10] FAILED tests/test_user_profile.py:61: TestUserProfile.test_that_user_can_edit_profile 
08:16:41 
08:16:41 =================================== FAILURES ===================================
08:16:41 _______________ TestUserProfile.test_that_user_can_edit_profile ________________
08:16:41 [gw10] darwin -- Python 2.7.5 /Users/Shared/Jenkins/Home/jobs/oneanddone.stage.saucelabs/workspace/.env/bin/python
08:16:41 self = <tests.test_user_profile.TestUserProfile instance at 0x101381ef0>
08:16:41 mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x101381f80>
08:16:41 existing_user = {'username': u'alexis179323', 'profile': {'username': u'alexis179323', 'privac...alexis179323@personatestuser.org', 'password': u'qxX7MLf0jU1e7MLU', 'id': 9013}
08:16:41 
08:16:41     @pytest.mark.credentials
08:16:41     def test_that_user_can_edit_profile(self, mozwebqa, existing_user):
08:16:41         home_page = HomePage(mozwebqa)
08:16:41         home_page.go_to_page()
08:16:41         Assert.false(home_page.is_user_logged_in)
08:16:41     
08:16:41         logged_in_home_page = home_page.login(existing_user, 'home_page')
08:16:41         Assert.true(logged_in_home_page.is_the_current_page)
08:16:41         Assert.true(logged_in_home_page.is_user_logged_in)
08:16:41     
08:16:41         # click user profile details link
08:16:41         user_profile_details_page = logged_in_home_page.header.click_user_profile_details()
08:16:41         Assert.true(user_profile_details_page.is_the_current_page)
08:16:41     
08:16:41         # click edit profile button
08:16:41         user_profile_edit_page = user_profile_details_page.click_edit_profile_button()
08:16:41         Assert.true(user_profile_edit_page.is_the_current_page)
08:16:41         Assert.equal(existing_user['profile']['name'], user_profile_edit_page.display_name)
08:16:41         Assert.equal(existing_user['profile']['username'], user_profile_edit_page.username)
08:16:41     
08:16:41         # change display name and username
08:16:41         new_display_name = re.sub(r'[\W_]+', '', existing_user['profile']['name'] + str(time.time()))
08:16:41         new_username = re.sub(r'[\W_]+', '', existing_user['profile']['username'] + str(time.time()))
08:16:41         user_profile_edit_page.enter_name(new_display_name)
08:16:41         user_profile_edit_page.enter_username(new_username)
08:16:41         home_page = user_profile_edit_page.click_save_button()
08:16:41     
08:16:41 >       Assert.true(home_page.is_the_current_page)
08:16:41 
08:16:41 tests/test_user_profile.py:88: 
08:16:41 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
08:16:41 
08:16:41 self = <pages.home.HomePage object at 0x101373410>
08:16:41 
08:16:41     @property
08:16:41     def is_the_current_page(self):
08:16:41         WebDriverWait(self.selenium, 10).until(lambda s: self.selenium.title)
08:16:41         Assert.equal(self.selenium.title, self._page_title,
08:16:41 >                    'Expected page title: %s. Actual page title: %s' % (self._page_title, self.selenium.title))
08:16:41 
08:16:41 pages/page.py:36: 
08:16:41 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
08:16:41 
08:16:41 self = <class 'unittestzero.Assert'>
08:16:41 first = u'User profile for mozwebqatestuser142116579708 | Mozilla One and Done'
08:16:41 second = 'Mozilla One and Done'
08:16:41 msg = u'Expected page title: Mozilla One and Done. Actual page title: User profile for mozwebqatestuser142116579708 | Mozilla One and Done'
08:16:41 
08:16:41     @classmethod
08:16:41     def equal(self, first, second, msg=None):
08:16:41         """
08:16:41             Asserts that 2 elements are the same
08:16:41     
08:16:41             :Args:
08:16:41              - First object to be tested
08:16:41              - Second object to be tested
08:16:41              - Message that will be printed if it fails
08:16:41             """
08:16:41 >       assert first == second, msg
08:16:41 E       AssertionError: Expected page title: Mozilla One and Done. Actual page title: User profile for mozwebqatestuser142116579708 | Mozilla One and Done
08:16:41 
08:16:41 .env/lib/python2.7/site-packages/unittestzero.py:55: AssertionError
08:16:41 ------------------------------- pytest-mozwebqa --------------------------------
08:16:41 Failing URL: https://oneanddone.allizom.org/en-US/profile/

### PR DESCRIPTION
**Pull request by bobsilverberg**
This fixes `test_that_user_can_edit_profile` which has been failing since a bug was pushed for One and Done.

Adhoc running at http://selenium.qa.mtv2.mozilla.com:8080/job/oneanddone.stage.adhoc/1/

@stephendonner and/or @m8ttyB r?

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Oops, that first adhoc failed, but this one should be good: http://selenium.qa.mtv2.mozilla.com:8080/job/oneanddone.stage.adhoc/2/

**Comment by rbillings** [comment]
Ran tests locally, both login and user profile tests pass. LGTM.

File: pages/base.py [hunk 1]
```diff
@@ -31,9 +31,12 @@ def login(self, user, expectation):
         return self.expected_page(expectation)
 
     def expected_page(self, expectation):
```
Comment (m8ttyB) [inline pages/base.py]: Interesting pattern/method.

**Comment by m8ttyB** [comment]
lgtm -- merging

cativator

**Pull request merged**